samples/csharp/sharedcontent/console/speech_recognition_samples.cs

```diff
@@ -943,5 +943,42 @@ public static async Task KeywordRecognizer()
                 }
             }
         }
+
+        private static async Task<RecognitionResult> RecognizeOnceAsyncInternal(string key, string region)
+        {
+            RecognitionResult recognitionResult = null;
+            var config = SpeechConfig.FromSubscription(key, region);
+
+            // Creates a speech recognizer using file as audio input.
+            using (var audioInput = AudioConfig.FromWavFileInput(@"whatstheweatherlike.wav"))
+            {
+                using (var recognizer = new SpeechRecognizer(config, audioInput))
+                {
+                    recognitionResult = await recognizer.RecognizeOnceAsync().ConfigureAwait(false);
+                }
+            }
+
+            return recognitionResult;
+        }
+
+        // Speech recognition with backup subscription region.
+        public static async Task RecognitionOnceWithFileAsyncSwitchSecondaryRegion()
+        {
+            // Create a speech resource with primary subscription key and service region.
+            // Also create a speech resource with secondary subscription key and service region 
+            RecognitionResult recognitionResult = await RecognizeOnceAsyncInternal("PrimarySubscriptionKey", "PrimarySubscriptionRegion");
+            if (recognitionResult.Reason == ResultReason.Canceled)
+            {
+                CancellationDetails details = CancellationDetails.FromResult(recognitionResult);
+                if (details.ErrorCode == CancellationErrorCode.ConnectionFailure
+                    || details.ErrorCode == CancellationErrorCode.ServiceUnavailable
+                    || details.ErrorCode == CancellationErrorCode.ServiceTimeout)
```

Comment:
It sure would be nice if the service error was specific to fail over, and perhaps even recommended a region to failover to... but that's just my wish, don't know if the front end service can do that yet failover redirect...

Reply:
Currently no. Because the issue is the service itself is flaky and network issue, so it will not even reply to any request